### ISSUE TITLE
MimiCodec Checkpoints?

### ISSUE THREAD

**Issue by Respaired** [yangdongchao/RSTnet#17]
Hi, Thanks for creating this much needed work.

do you have any fully trained checkpoint for Mimi so we can play around with it? based on what some other people have said in the other issue, I want to make sure this is working before fully investing on it. I appreciate it if you can share the weights. Thanks!

**Comment by kadirnar** [comment]
> Hi, Thanks for creating this much needed work.
> 
> do you have any fully trained checkpoint for Mimi so we can play around with it? based on what some other people have said in the other issue, I want to make sure this is working before fully investing on it. I appreciate it if you can share the weights. Thanks!

Isn't this the MimiCodec model?

HF: https://huggingface.co/kyutai/mimi
Doc: https://huggingface.co/docs/transformers/v4.48.2/en/model_doc/mimi

**Comment by Respaired** [comment]
it is an implementation of that.
afaik the official code doesn't come with a training code.

**Comment by Respaired** [comment]
I tried to train a 44.1khz version but it seems the loss is diverging or stays roughly in the avenue of ~7. , even with mel_loss it didn't help at all. also heavy signs of discriminator overpowering the generator; though I'm not sure how much any of these are natural.

```
epoch: 0, iter: 200, Train/G_adv_mfd: 0.5041691064834595, Train/G_fm_mfd: 0.01733989119529724, Train/G_mg_full: 1.798746943473816, Train/G_mg_sub: 1.2328746318817139, Train/G_sc_full: 0.9183715581893921, Train/G_sc_sub: 0.9148630499839783, Train/codec_loss: 0.9547044634819031, Train/g_loss: 6.670527458190918, Train/D_mfd: 0.36183854937553406, Train/d_loss: 0.36183854937553406,
epoch: 0, iter: 300, Train/G_adv_mfd: 0.6964106559753418, Train/G_fm_mfd: 0.02546767294406891, Train/G_mg_full: 2.740407943725586, Train/G_mg_sub: 2.0224318504333496, Train/G_sc_full: 0.9560408592224121, Train/G_sc_sub: 0.952859103679657, Train/codec_loss: 0.6346454620361328, Train/g_loss: 8.512149810791016, Train/D_mfd: 0.1569914072751999, Train/d_loss: 0.1569914072751999,
epoch: 0, iter: 400, Train/G_adv_mfd: 0.37448638677597046, Train/G_fm_mfd: 0.01987968534231186, Train/G_mg_full: 2.519442319869995, Train/G_mg_sub: 1.870812177658081, Train/G_sc_full: 1.1123206615447998, Train/G_sc_sub: 1.165839433670044, Train/codec_loss: 0.32117143273353577, Train/g_loss: 7.7616658210754395, Train/D_mfd: 0.416095107793808, Train/d_loss: 0.416095107793808,
epoch: 0, iter: 500, Train/G_adv_mfd: 0.47828489542007446, Train/G_fm_mfd: 0.020271751284599304, Train/G_mg_full: 2.320164680480957, Train/G_mg_sub: 1.685533046722412, Train/G_sc_full: 1.0114059448242188, Train/G_sc_sub: 1.0062286853790283, Train/codec_loss: 0.31539273262023926, Train/g_loss: 7.222444534301758, Train/D_mfd: 0.33321768045425415, Train/d_loss: 0.33321768045425415,
epoch: 0, iter: 600, Train/G_adv_mfd: 0.5035558938980103, Train/G_fm_mfd: 0.021610230207443237, Train/G_mg_full: 2.312077760696411, Train/G_mg_sub: 1.6743621826171875, Train/G_sc_full: 0.9689353704452515, Train/G_sc_sub: 0.9679244756698608, Train/codec_loss: 0.31444278359413147, Train/g_loss: 7.1735029220581055, Train/D_mfd: 0.28778308629989624, Train/d_loss: 0.28778308629989624,
epoch: 0, iter: 700, Train/G_adv_mfd: 0.5831171274185181, Train/G_fm_mfd: 0.02342063784599304, Train/G_mg_full: 2.4086616039276123, Train/G_mg_sub: 1.776350975036621, Train/G_sc_full: 1.0009715557098389, Train/G_sc_sub: 0.9993195533752441, Train/codec_loss: 0.31416967511177063, Train/g_loss: 7.551003456115723, Train/D_mfd: 0.21777915954589844, Train/d_loss: 0.21777915954589844,
epoch: 0, iter: 800, Train/G_adv_mfd: 0.7183656692504883, Train/G_fm_mfd: 0.024400603771209717, Train/G_mg_full: 2.222592353820801, Train/G_mg_sub: 1.5634613037109375, Train/G_sc_full: 1.0008699893951416, Train/G_sc_sub: 0.9964038133621216, Train/codec_loss: 0.3140026926994324, Train/g_loss: 7.303707599639893, Train/D_mfd: 0.16570784151554108, Train/d_loss: 0.16570784151554108,
epoch: 0, iter: 900, Train/G_adv_mfd: 0.6866812109947205, Train/G_fm_mfd: 0.025209718942642213, Train/G_mg_full: 2.307281017303467, Train/G_mg_sub: 1.615112066268921, Train/G_sc_full: 0.9583813548088074, Train/G_sc_sub: 0.9580941200256348, Train/codec_loss: 0.3138973116874695, Train/g_loss: 7.34364128112793, Train/D_mfd: 0.16039498150348663, Train/d_loss: 0.16039498150348663,
epoch: 0, iter: 1000, Train/G_adv_mfd: 0.7539112567901611, Train/G_fm_mfd: 0.028483015298843384, Train/G_mg_full: 2.299712657928467, Train/G_mg_sub: 1.6930475234985352, Train/G_sc_full: 0.9812198877334595, Train/G_sc_sub: 0.980402410030365, Train/codec_loss: 0.3138192296028137, Train/g_loss: 7.591773509979248, Train/D_mfd: 0.10769589990377426, Train/d_loss: 0.10769589990377426,
epoch: 0, iter: 1100, Train/G_adv_mfd: 0.7663915157318115, Train/G_fm_mfd: 0.030347862839698793, Train/G_mg_full: 2.480379581451416, Train/G_mg_sub: 1.8306961059570312, Train/G_sc_full: 1.0386903285980225, Train/G_sc_sub: 1.032525897026062, Train/codec_loss: 0.31364867091178894, Train/g_loss: 8.069289207458496, Train/D_mfd: 0.09313153475522995, Train/d_loss: 0.09313153475522995,
```

this is my config, though the default one was kind of leading to the same result in the first 2K steps.

```config
generator:
  name: MimiCodec
  config:
    n_filters: 64
    sample_rate: 44100
    encoder_rates: [4, 5, 6, 8] 
    compress: 2 # last compress
    latent_dim: 256
    codebook_size: 1024
    codebook_dim: 128
    rvq_layers: 8
    num_heads: 8
    num_layers: 8
    layer_scale: 0.01
    context: 500
    dim_feedforward: 1024
    semantic_feature_dim: 1024 # different semantic model will produce different results
  ```

@yangdongchao 

I understand you've a lot on your plate, but may I ask for your help. this work is really interesting (if it works); I'd appreciate it if you help me out a bit here.

update: after 31K steps on the default settings (exact same config and SR in this repository), the g_loss is now 11~. 
yup, either i'm doing something wrong, or the repo is just broken.
